### PR TITLE
Ledger API: add `buf` checks [KVL-1045]

### PR DESCRIPTION
**Pull request by fabiotudone-da**
Depends on #10624.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: fmt.sh [hunk 1]
```diff
@@ -156,6 +156,7 @@ run scalafmt "${scalafmt_args[@]:-}"
 run bazel run "$buildifier_target"
 
 # Check for breaking proto changes.
+run buf mod update
```
Comment (SamirTalwar-DA) [inline fmt.sh:159]: What does this command do? I can't find the relevant docs.

Comment (fabiotudone-da) [reply fmt.sh:159]: Found it [here](https://docs.buf.build/tour/add-a-dependency/#depend-on-googleapis).

Comment (fabiotudone-da) [reply fmt.sh:159]: This is needed because `completion.proto` depends on gRPC's `Status` and that's in `googleapis`.

**Review by miklos-da — APPROVED** [review]

**Pull request merged**